nlpsandboxclient/api_client.py

```diff
@@ -86,120 +98,182 @@ def _build_uri(self, uri, endpoint=None):
+                          annotation_storeid=annotation_storeid,
+                          id=annotationid,
+                          **annotation)
 
     def create_annotation(self, datasetid: str, annotation_storeid: str,
-                          annotation: dict):
+                          annotation: dict) -> Annotation:
         """Create an annotation"""
-        return self.rest_post(
+        annotation = self.rest_post(
             f"/datasets/{datasetid}/annotationStores/"
             f"{annotation_storeid}/annotations",
-            body=json.dumps(annotation)
+            body=annotation
         )
+        return Annotation(datasetid=datasetid,
+                          annotation_storeid=annotation_storeid,
+                          id=os.path.basename(annotation['name']),
+                          **annotation)
 
-    def list_fhir_stores(self, datasetid: str):
+    def list_fhir_stores(self, datasetid: str) -> Iterator[FhirStore]:
         """List the FHIR stores in a dataset"""
-        return self.rest_get_paginated(f"/datasets/{datasetid}/fhirStores")
+        fhir_stores = self.rest_get_paginated(
+            f"/datasets/{datasetid}/fhirStores"
+        )
+        for fhir_store in fhir_stores:
+            yield FhirStore(datasetid=datasetid,
+                            id=os.path.basename(fhir_store['name']),
+                            **fhir_store)
 
-    def get_fhir_store(self, datasetid: str, fhir_storeid: str):
+    def get_fhir_store(self, datasetid: str, fhir_storeid: str) -> FhirStore:
         """Get a FHIR store"""
-        return self.rest_get(
+        fhir_store = self.rest_get(
             f"/datasets/{datasetid}/fhirStores/{fhir_storeid}"
         )
+        return FhirStore(datasetid=datasetid, id=fhir_storeid,
+                         **fhir_store)
 
-    def create_fhir_store(self, datasetid: str, fhir_storeid: str):
+    def create_fhir_store(self, datasetid: str,
+                          fhir_storeid: str) -> FhirStore:
         """Create a FHIR store"""
-        return self.rest_post(
+        fhir_store = self.rest_post(
             f"/datasets/{datasetid}/fhirStores?fhirStoreId={fhir_storeid}",
-            body=json.dumps({})
+            body={}
         )
+        return FhirStore(datasetid=datasetid, id=fhir_storeid,
+                         **fhir_store)
 
-    def list_clinical_notes(self, datasetid: str, fhir_storeid: str):
+    def list_clinical_notes(self, datasetid: str,
+                            fhir_storeid: str) -> Iterator[Note]:
         """List clinical notes in a FHIR store"""
-        return self.rest_get_paginated(
+        notes = self.rest_get_paginated(
             f"/datasets/{datasetid}/fhirStores/{fhir_storeid}/fhir/Note"
         )
+        for note in notes:
+            yield Note(datasetid=datasetid, fhir_storeid=fhir_storeid, **note)
 
     def get_clinical_note(self, datasetid: str, fhir_storeid: str,
```

Comment:
The schema is called `Note`, so let's rename `list_clinical_notes` to `list_notes`.